### DISCUSSION TITLE
Welcome to `huh?` Discussions!

### DISCUSSION THREAD

**Discussion by maaslalani** [charmbracelet/huh#46] [Announcements]
## Welcome!

Please feel free to:

  * Ask questions.
  * Share ideas.

**Comment by GianniBYoung** [comment]
Can someone clarify what the behavior of `NewInput().CharLimit()` is? 

I have the following code and it allows me to exceed the character limit. Is this intended behavior?

I also tested  `NewText().CharLimit()` and that method stopped me from typing past the limit.

```
package main

import (
	"fmt"

	"github.com/charmbracelet/huh"
)

var (
	name string
)

func main() {
	form := huh.NewForm(huh.NewGroup(
		huh.NewInput().Title("What is your name?").CharLimit(30).Value(&name),
	))

	form.Run()
	fmt.Println(name)
}
```

**Reply by maaslalani** [reply]
Hey @GianniBYoung, thanks for pointing this out. It was a bug, I have fixed it in https://github.com/charmbracelet/huh/pull/79

**Reply by GianniBYoung** [reply]
Thanks!
